packages/client/test/rpc/rpc.spec.ts

```diff
@@ -7,53 +8,48 @@ import { METHOD_NOT_FOUND } from '../../src/rpc/error-code'
 import { baseRequest, closeRPC, createClient, createManager, params, startRPC } from './helpers'
 
 import type { TAlgorithm } from 'jwt-simple'
+import type { AddressInfo } from 'net'
 
 const request = require('supertest')
 
 const jwtSecret = randomBytes(32)
 
 describe('JSON-RPC call', () => {
-  it('without Content-Type header', () => {
+  it('without Content-Type header', async () => {
     const server = startRPC({})
+    const rpc = Client.http({ port: (server.address()! as AddressInfo).port })
     const req = 'plaintext'
-
-    request(server)
-      .post('/')
-      .send(req)
-      .expect(415)
-      .end((err: any) => {
-        closeRPC(server)
-        assert.notOk(err)
-      })
+    const res = await rpc.request(req, [])
+    assert.equal(res.error.code, -32601)
   })
 
-  it('auth protected server without any auth headers', () => {
+  it('auth protected server without any auth headers', async () => {
     const server = startRPC({}, undefined, { jwtSecret })
+    const rpc = Client.http({ port: (server.address()! as AddressInfo).port })
     const req = 'plaintext'
-
-    request(server)
-      .post('/')
-      .send(req)
-      .expect(401)
-      .end((err: any) => {
-        closeRPC(server)
-        assert.notOk(err)
-      })
+    try {
+      await rpc.request(req, [])
+      assert.notOk(undefined, 'should error when request not authenticated by JWT')
```

Comment:
Short nit: I think this can be shortened to `assert.fail('should error...')`

Reply:
Good catch.